cartography/intel/github/repos.py

```diff
@@ -162,14 +164,42 @@ def _get_repo_collaborators_for_multiple_repos(
             result[repo_url] = []
             continue
 
-        collab_users = []
-        collab_permission = []
-        collaborators = _get_repo_collaborators(token, api_url, org, repo_name, affiliation)
-        # nodes and edges are expected to always be present given that we only call for them if totalCount is > 0
-        for collab in collaborators.nodes:
-            collab_users.append(collab)
-        for perm in collaborators.edges:
-            collab_permission.append(perm['permission'])
+        collab_users: List[dict[str, Any]] = []
+        collab_permission: List[str] = []
+
+        def get_repo_collaborators_inner_func(
+                org: str,
+                api_url: str,
+                token: str,
+                repo_name: str,
+                affiliation: str,
+                collab_users: List[dict[str, Any]],
+                collab_permission: List[str],
+        ) -> None:
+            logger.info(f"Loading {affiliation} collaborators for repo {repo_name}.")
```

Comment:
nit thing: i'm not entirely sure the need and effect of having this func defined inside a for-loop. I don't immediately see the need for it since all the params are passed in it seems like. If so, we can move it out to the package level for clarity